Failed to build on FreeBSD

```
$ cmake -S . -B build
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The CXX compiler identification is Clang 18.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at cmake/CMakeUtils.cmake:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
Call Stack (most recent call first):
  CMakeLists.txt:6 (include)


CMake Deprecation Warning at cmake/CMakeCompiler.cmake:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
Call Stack (most recent call first):
  CMakeLists.txt:7 (include)


CMake Deprecation Warning at cmake/CMakePlatforms.cmake:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
Call Stack (most recent call first):
  CMakeLists.txt:8 (include)


CMake Error at cmake/CMakePlatforms.cmake:28 (message):
  Unknown platform FreeBSD!
Call Stack (most recent call first):
  CMakeLists.txt:8 (include)


-- Configuring incomplete, errors occurred!
```

--- a/.github/workflows/build_pull_request.yml
+++ b/.github/workflows/build_pull_request.yml
@@ -325,11 +325,11 @@ jobs:
         with:
           args: 'python3 make.py -ci -compiler ${{ matrix.compiler }} -config Release -cpu x64 -avx -unit_test'
 
-  osx-12:
-    runs-on: macos-12
+  osx-13:
+    runs-on: macos-13
     strategy:
       matrix:
-        compiler: [xcode13]
+        compiler: [xcode14, xcode15]
     steps:
       - name: Git checkout
         uses: actions/checkout@v4
@@ -370,7 +370,7 @@ jobs:
     runs-on: macos-14
     strategy:
       matrix:
-        compiler: [xcode14, xcode15]
+        compiler: [xcode15, xcode16]
     steps:
       - name: Git checkout
         uses: actions/checkout@v4

--- a/.github/workflows/build_pull_request.yml
+++ b/.github/workflows/build_pull_request.yml
@@ -329,7 +329,7 @@ jobs:
     runs-on: macos-13
     strategy:
       matrix:
-        compiler: [xcode14, xcode15]
+        compiler: [xcode14]
     steps:
       - name: Git checkout
         uses: actions/checkout@v4

--- a/.github/workflows/msvc-analysis.yml
+++ b/.github/workflows/msvc-analysis.yml
@@ -42,7 +42,7 @@ jobs:
       - name: Upload SARIF to GitHub
         # Can only upload SARIF if we don't come from a fork
         if: ${{ needs.condition-check.outputs.ok == 'true' }}
-        uses: github/codeql-action/upload-sarif@v2
+        uses: github/codeql-action/upload-sarif@v3
         with:
           sarif_file: ${{ steps.run-analysis.outputs.sarif }}
       - name: Upload SARIF as an Artifact

--- a/cmake/CMakePlatforms.cmake
+++ b/cmake/CMakePlatforms.cmake
@@ -28,7 +28,7 @@ elseif(${UNIX})
 	set(PLATFORM_UNIX 1)
 	set(PLATFORM_NAME "UNIX")
 else()
-	message(FATAL_ERROR "Unknown platform ${CMAKE_SYSTEM_NAME}!")
+	set(PLATFORM_NAME ${CMAKE_SYSTEM_NAME})
 endif()
 
 message(STATUS "Detected platform: ${PLATFORM_NAME}")

--- a/tools/setup_osx_compiler.sh
+++ b/tools/setup_osx_compiler.sh
@@ -5,7 +5,7 @@ COMPILER=$1
 
 # See Github hosted runners:
 # macos-14: https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md
-#   xcode 14.3.1, 15.0.1, 15.1, 15.2, 15.3, 15.4, 16.0
+#   xcode 15.0.1, 15.1, 15.2, 15.3, 15.4, 16.1, 16.2
 # macos-13: https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md
 #   xcode 14.1, 14.2, 14.3.1, 15.0.1, 15.1, 15.2
 # macos-12: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
@@ -28,9 +28,9 @@ elif [[ $COMPILER == xcode13 ]]; then
 elif [[ $COMPILER == xcode14 ]]; then
     XCODE_PATH="/Applications/Xcode_14.3.1.app"
 elif [[ $COMPILER == xcode15 ]]; then
-    XCODE_PATH="/Applications/Xcode_15.2.app"
+    XCODE_PATH="/Applications/Xcode_15.4.app"
 elif [[ $COMPILER == xcode16 ]]; then
-    XCODE_PATH="/Applications/Xcode_16.0.app"
+    XCODE_PATH="/Applications/Xcode_16.2.app"
 fi
 
 # Select our XCode version

